core/lib/spree/app_configuration.rb

```diff
@@ -155,6 +155,18 @@ class AppConfiguration < Preferences::Configuration
     #   @return [Regexp] Regex to be used in email validations, for example in Spree::EmailValidator
     preference :default_email_regexp, :regexp, default: URI::MailTo::EMAIL_REGEXP
 
+    # @!attribute [rw] extra_taxon_validations
+    #   Use extra validations on Taxons in 3.4.0, but default to false so stores can
+    #   upgrade and then fix any now invalid Taxons before enabling.
+    #   @return [Boolean]
+    versioned_preference :extra_taxon_validations, :boolean, initial_value: false, boundaries: { "3.4.0" => true }
```

Comment:
```suggestion
    versioned_preference :extra_taxon_validations, :boolean, initial_value: false, boundaries: { "3.4.0.dev" => true }
```

You can use the current version in `core/lib/spree/core/version.rb` to fix CI :slightly_smiling_face: